### PR TITLE
feat: add membership level to user / person data

### LINKED ISSUE #18330: Instrumentation for membership level
this is a request for posthog-specific instrumentation

we don’t have `membership_level` anywhere in event/person data, afaik it only lives in postgres. Aside from the fact that this unlocks a whole bunch of interesting analyses around product usage behavior within the app, it is also something that informs our outreach, but we currently have to open admin panel to figure this out. It would be very useful to have this on the posthog person, along with an event when membership level changes. At the very least we could add a property to the already existing metric for this capturing the permission level so we at least have the initial permission level. I believe [here](https://github.com/PostHog/posthog/blob/50811b4df408b8b8be33862c5e56c416bd1260d1/posthog/event_usage.py#L157) is the relevant file.

## Describe the solution you'd like

membership_level to be added as a property to the existing event for invites (ideally to the event attributed to the _invitee_, and for it to be $set, and for an additional event in case someone changes the users membership level in the future (again, to be attributed to the _invitee_).

## Describe alternatives you've considered

look in postgres via admin every time 🕐 🕐 🕐 

## Additional context

https://posthog.slack.com/archives/C02E3BKC78F/p1698870804030639

### PR DESCRIPTION
**Pull request by raquelmsmith**
## Problem

[CS would find it useful](https://posthog.slack.com/archives/C02E3BKC78F/p1698870804030639) to know what membership level a person has. 

Closes https://github.com/PostHog/posthog/issues/18330

## Changes

- Sends the person's current org membership level in their person analytics metadata
- Sends an event when a level changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

added tests

### CODE REVIEW

**Comment by raquelmsmith** [comment]
@Twixes you get this review since your name is all over the blame for these files 😄

File: posthog/event_usage.py [hunk 1]
```diff
@@ -196,6 +196,27 @@ def report_bulk_invited(
...
+    previous_level: int,
+) -> None:
+    """
+    Triggered after a user's membership level in an organization is changed.
+    """
+    posthoganalytics.capture(
+        user.distinct_id,
+        "membership level changed",
+        properties={
+            "organization_id": str(organization.id),
```
Comment (Twixes) [inline posthog/event_usage.py:212]: Isn't this redundant with the `organization` _group_ already associated with the event?

**Review by Twixes — APPROVED** [review]
Looks good.
How's this going to be used in analysis BTW? Wondering because there's no way of setting this on a person _or_ an `organization` group, given that an organization membership is a _relationship_ between a person and a group

**Comment by raquelmsmith** [comment]
> Wondering because there's no way of setting this on a person or an organization group, given that an organization membership is a relationship between a person and a group

Can you explain what you mean here? I did set it on the person, but the value changes when they change their org, which CS said they are fine with.

**Review by xrdt — APPROVED** [review]

**Comment by Twixes** [comment]
@raquelmsmith Hm, there's no `$set` in `properties` here. How does this set the property on the person otherwise?

BTW will it be safe to close #18330 once this is merged?

**Comment by Twixes** [comment]
What I mean by "there's no way of setting this on a person" is that a user might be (not often) an owner in one org and a plain member in another, which can't be reflected with a scalar `membership_level` person property.

**Comment by camerondeleone** [comment]
I see the complexity now and think it's fine to leave this off person properties. We can reconstruct what we need from a hogql query with just the events. Or change it to an array of orgs / membership, but that seems more complex because that makes the `$set` require context from other orgs and maybe another db lookup etc. 

Is the problem clear enough? I want to know who the owner and admin are without looking in admin, because they are the ones that should be contacted first... Open to suggestions.

**Comment by raquelmsmith** [comment]
> Hm, there's no $set in properties here. How does this set the property on the person otherwise?

I put the property in `user.get_analytics_metadata()`, which is called in this task from here: https://github.com/PostHog/posthog/blob/2d909a421dca88f7d4111ed994a579ad7e5ae1d2/posthog/api/user.py#L244

Which uses `$set` here, but maybe I am missing something? 

https://github.com/PostHog/posthog/blob/06e993818d014eded0d4168ffe1080f0fce93525/posthog/tasks/user_identify.py#L13C2-L13C2

> BTW will it be safe to close https://github.com/PostHog/posthog/issues/18330 once this is merged?

Yep! will update description.

**Pull request merged**